### DISCUSSION TITLE
How to obtain the crate that the LSP is going to import

### DISCUSSION THREAD

**Discussion by b1nhack** [mrcjkb/rustaceanvim#451] [Q&A]
nvim-cmp provides auto-completion support. During development, there are often many completion items with the same name, making it difficult to distinguish them. Once a completion item is selected, the crate will be automatically imported, which I suspect is handled by the LSP. If I can obtain the crate information that the completion item is going to import, such as `std::io::write`, it will be much easier to distinguish completion items with the same name.

**Comment by mrcjkb** [comment]
Hey 👋 

You're probably better off asking in the rust-analyzer or nvim-cmp discussions.

This plugin doesn't provide any completion features itself. See [the FAQ](https://github.com/mrcjkb/rustaceanvim?tab=readme-ov-file#how-to-enable-auto-completion).
